storage/remote/queue_manager_test.go

```diff
@@ -82,3 +109,40 @@ func TestSampleDelivery(t *testing.T) {
 
 	c.waitForExpectedSamples(t)
 }
+
+func TestSpawnNotMoreThanMaxConcurrentSendsGoroutines(t *testing.T) {
+	// `maxSamplesPerSend*maxConcurrentSends + 1` samples should be consumed by
+	//  goroutines, `maxSamplesPerSend` should be still in the queue
+	n := maxSamplesPerSend*maxConcurrentSends + maxSamplesPerSend*2
+
+	samples := make(model.Samples, 0, n)
+	for i := 0; i < n; i++ {
+		samples = append(samples, &model.Sample{
+			Metric: model.Metric{
+				model.MetricNameLabel: "test_metric",
+			},
+			Value: model.SampleValue(i),
+		})
+	}
+
+	c := NewTestBlockedStorageClient()
+	m := NewStorageQueueManager(c, n)
+
+	go m.Run()
+
+	for _, s := range samples {
+		m.Append(s)
+	}
+
+	for i := 0; i < maxConcurrentSends; i++ {
+		c.getData <- true // wait while all goroutines are spawned
```

Comment:
Comments should be full sentences. Start with a capital letter and end with a full stop.